gwtp-core/gwtp-clients-common/src/main/java/com/gwtplatform/common/client/ClientUrlUtils.java

```diff
@@ -23,6 +23,9 @@
  * Helper implementation which wraps calls to code which require a running GWT environment and make testing slow.
  */
 public class ClientUrlUtils implements UrlUtils {
+    // TODO: Review what should be encoded in various parameter types. Don't rely on GWT's code.
```

Comment:
You can't call the GWT URL encode/ decode from the server.  So we use the UrlUtils interface in shared code instead and create a Server/Client implementation.